### PR TITLE
First pass at page template list of items.

### PR DESCRIPTION
**Pull request by digithed**
Addresses #8 

## Details
- First pass at a page template. Created a story called "Shell". Took a look at Nishant's personal site repo for creating a page with a list of items for starter code. Added cva functions and other tweaks to generalize it.
- Created light and dark modes for the template

## How to test
https://storybook-ejum3kycv-ninjha01.vercel.app/?path=/story/ui-shell--shell-story

### CODE REVIEW

File: src/components/ui/Shell/Shell.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Shell } from "./Shell";
+
+export default {
+  title: "UI/Shell",
+  component: Shell,
+} as ComponentMeta<typeof Shell>;
+
+const Template: ComponentStory<typeof Shell> = (args) => {
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.stories.tsx:9]: don't think we actually need variants on this story. Just dark mode selector should be good! Can remove args

File: src/components/ui/Shell/Shell.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
+import { cva, VariantProps } from "class-variance-authority";
+import ReactMarkdown from "react-markdown";
+import remarkGfm from "remark-gfm";
+
+const shellStylesOuter = cva(
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:5]: I don't think we need any variants for this page template - you can omit cva for now. Just use the primary style you've set up here and drop the secondary one.

File: src/components/ui/Shell/Shell.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  "fixed inset-0 mx-auto mx-auto flex w-full max-w-4xl justify-center rounded-lg px-4 px-4 sm:px-6 sm:px-6 md:px-8 md:px-8",
+  {
+    variants: {
+      intent: {
+        primary:
+          "border-solid border-2 border-zinc-200 bg-white dark:bg-zinc-900",
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:11]: For the future, anything that's shared between all variants can be yanked up to the "invariant" string on L6

File: src/components/ui/Shell/Shell.tsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+  },
+  defaultVariants: {
+    intent: "primary",
+  },
+});
+
+const hrefStyles = cva("", {
+  variants: {
+    intent: {
+      primary: "text-blue-300 dark:text-blue-300",
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:64]: In general, don't use any colors other than `zinc` or `brand`. Not a hard and fast rule, but this way we can customize the colors easily in child projects

File: src/components/ui/Shell/Shell.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+          })}`}
+        >
+          {name}
+        </h2>
+        <div className="md:col-span-3">
+          <ul role="list" className="space-y-16">
+            {cards.map((item, index) => {
+              return (
+                <UIMockupCard
+                  key={index}
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:158]: In general, let's avoid using indexes as keys:
["We don’t recommend using indexes for keys if the order of items may change."](https://reactjs.org/docs/lists-and-keys.html)

File: src/components/ui/Shell/Shell.tsx [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+                  intent,
+                })} sm:text-5xl`}
+              >
+                About Me
+              </h1>
+              <div className={subHeaderStyles({ intent })}>
+                {"Some things about me."}
+              </div>
+            </header>
+            <UISection
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:244]: Let's have this component take a `sections` prop and then map those here

**Review by ninjha01 — CHANGES_REQUESTED** [review]
First round of feedback. Didn't get a chance to go super in depth but hopefully there's enough to chew on!

Thanks for the work!

File: src/components/ui/Shell/Shell.tsx (hunk 6 above)
Comment (ninjha01) [reply src/components/ui/Shell/Shell.tsx]: I was a bit vague, something like 
```
sections.map(section => { ...section stuff, cards: section.entries.map(entry => {...card stuff})})
```

File: src/components/ui/Shell/Shell.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
+import { cva, VariantProps } from "class-variance-authority";
+import ReactMarkdown from "react-markdown";
+import remarkGfm from "remark-gfm";
+
+const shellStylesOuter = cva(
+  "fixed inset-0 mx-auto mx-auto flex w-full h-full max-w-4xl border-solid border-2 border-zinc-200 bg-white dark:bg-zinc-900 justify-center rounded-lg px-4 px-4 sm:px-6 sm:px-6 md:px-8 md:px-8"
+);
+
+const shellStylesInner = cva(
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:9]: If we're not using any variants, cva is just extra overhead - let's throw these inline

File: src/components/ui/Shell/Shell.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+export const Shell = ({ sections, aboutMe }: ShellProps) => {
+  return (
+    <div>
+      <div className={shellStylesOuter({})} />
+      <div className="relative">
+        <main className={shellStylesInner({})}>
+          <div className="mx-auto max-w-2xl lg:max-w-5xl">
+            <header className="max-w-2xl">
+              <h1 className={textStylesHeader({})}>About Me</h1>
```
Comment (ninjha01) [inline src/components/ui/Shell/Shell.tsx:151]: Let's turn "About Me" into a `title` prop and rename `aboutMe` to `subtitle`

**Review by ninjha01 — DISMISSED** [review]
Just some small changes - looking good!

File: src/components/ui/Shell/Shell.tsx (hunk 6 above)
Comment (digithed) [reply src/components/ui/Shell/Shell.tsx]: Yea I pass an array of card objects to each `UISection` component and then map that array as individual `UIMockupCard` components. I think we are in agreement

**Comment by ninjha01** [comment]
Made some quick changes - if they look good/make sense LGTM!

**Pull request merged**